dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -484,11 +487,90 @@ private String getQuerySelect( TrackedEntityInstanceQueryParams params, boolean
         {
             select.append( ", TEI.trackedentityinstanceid AS teiid" );
         }
+
+        select.append( getOrderParamForSelectClause( params, select ) );
+
         select.append( SPACE );
 
         return select.toString();
     }
 
+    /**
+     * Gets all the static fields from the order parameters and appends them to
+     * the string builder, so they can be used in the query select clause
+     *
+     * @param params
+     * @param select current select statement
+     * @return all the static fields that are not yet present in the select
+     *         statement
+     */
+    private String getOrderParamForSelectClause( TrackedEntityInstanceQueryParams params, StringBuilder select )
```

Comment:
Please pass in a `String select`. Passing in a `StringBuilder` signals to me that you'll add to it. Since you are just interested in `select.toString()` pass in a `String`.